app/lib/userID.coffee

```diff
@@ -0,0 +1,46 @@
+User = require 'zooniverse/lib/models/user'
+currentUserID = null
+
+getClientOrigin = ->
+  eventualIP = new $.Deferred
+  $.get('https://api.ipify.org')
+  .then (ip) =>
+    eventualIP.resolve {ip: ip, address: ip}
+  .fail =>
+    eventualIP.resolve {ip: '?.?.?.?', address: '(anonymous)'}
+  eventualIP.promise()
+
+getNiceOriginString = (data) ->
+  if data.ip? && data.address?
+    if data.ip == '?.?.?.?'
+      "(anonymous)"
+    else if data.ip == data.address
+      "(#{ data.ip })"
+    else
+      "(#{ data.address } [#{ data.ip }])"
+  else
+    "(anonymous)"
+
+getUserIDorIPAddress = =>
```

Comment:
From what I can tell, this file's purpose is to reconcile a user's zooniverse ID (if it exists), with their IP address.

Wouldn't this be easier to do server-side and not require hitting a separate API?